diesel_cli/tests/migration_list.rs

```diff
@@ -0,0 +1,113 @@
+
+use regex::Regex;
+use chrono::UTC;
+use std::thread::sleep;
+use std::time::Duration;
+
+use support::{database, project};
+use diesel::migrations::TIMESTAMP_FORMAT;
+
+#[test]
+fn migration_list_lists_pending_applied_migrations() {
+    let p = project("migration_list_pending_applied")
+        .folder("migrations")
+        .build();
+    let db = database(&p.database_url());
+
+    p.command("setup").run();
+
+    p.create_migration("12345_create_users_table",
+                       "CREATE TABLE users ( id INTEGER )",
+                       "DROP TABLE users");
+
+    assert!(!db.table_exists("users"));
+
+    // finds unapplied migration
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    assert!(result.stdout().contains("[ ] 12345_create_users_table"));
+
+    let result = p.command("migration")
+        .arg("run")
+        .run();
+    assert!(result.is_success());
+    assert!(db.table_exists("users"));
+
+    // finds applied migration
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    assert!(result.stdout().contains("[X] 12345_create_users_table"));
+}
+
+#[test]
+fn migration_list_lists_migrations_ordered_by_timestamp() {
+    let p = project("migration_list_ordered")
+        .folder("migrations")
+        .build();
+
+    p.command("setup").run();
+
+    let tag1 = format!("{}_initial", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag1, "", "");
+
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    assert!(result.stdout().contains(&format!("[ ] {}", &tag1)));
+
+    sleep(Duration::from_millis(1100));
+
+    let tag2 = format!("{}_alter", UTC::now().format(TIMESTAMP_FORMAT));
+    p.create_migration(&tag2, "", "");
+
+    let result = p.command("migration")
+        .arg("list")
+        .run();
+    assert!(result.is_success(), "Result was unsuccessful {:?}", result);
+    let pattern = format!(r"(\s|.)*\[ \] {}(\s|.)*\[ \] {}(\s|.)*", &tag1, &tag2);
+    let expected = Regex::new(&pattern).unwrap();
+    assert!(expected.is_match(result.stdout()));
```

Comment:
Yes, using `.find` looks much better and makes more sense for the test with a bunch of `tags`! I'll change that up later today.